meshroom/core/graph.py

```diff
@@ -1015,17 +1023,50 @@ def save(self, filepath=None, setupProjectFile=True):
             for p in usedNodeTypes
         }
 
-        data = {
-            Graph.IO.Keys.Header: self.header,
-            Graph.IO.Keys.Graph: self.toDict()
-        }
+        data = {}
+        if template:
+            self.header["template"] = True
+            data = {
+                Graph.IO.Keys.Header: self.header,
+                Graph.IO.Keys.Graph: self.getModifiedKeys()
+            }
+        else:
+            data = {
+                Graph.IO.Keys.Header: self.header,
+                Graph.IO.Keys.Graph: self.toDict()
+            }
 
         with open(path, 'w') as jsonFile:
             json.dump(data, jsonFile, indent=4)
 
         if path != self._filepath and setupProjectFile:
             self._setFilepath(path)
 
+    def getModifiedKeys(self):
```

Comment:
getModifiedKeys(self) => getNonDefaultAttributes(self)

We are not returning only keys (myDict.keys(), myDict.values()).

Reply:
Done